src/test/java/org/elasticsearch/test/disruption/SuspendDisruption.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.test.disruption;
+
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * Suspends all threads on the specified node. Note: this class isn't thread safe, so use it from the same thread
```

Comment:
I'm a bit worried about this..  can we make it use atomic reference to the suspended threads? 

Also, I think we should flip the class hierarchy, make this class the base of the current LongGCDisruption.  We should call this one LongGCDisruption and the other one IntermittentLongGCDisruption (or something like this). This class does all the heavy lifting and the IntermittentLongGCDisruption adds the "doing it in the background" aspect. Makes sense?